test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -317,3 +317,480 @@ test.describe( 'Navigation block', () => {
+		const firstMenuItemAnchor = listView.getByRole( 'link', {
+			name: 'Top Level Item 1',
+			includeHidden: true,
+		} );
+		await firstMenuItemAnchor.click();
+
+		// Get the settings panel.
+		const blockSettings = page.getByRole( 'region', {
+			name: 'Editor settings',
+		} );
+
+		await expect( blockSettings ).toBeVisible();
+
+		await expect(
+			blockSettings.getByRole( 'heading', {
+				name: 'Page Link',
+			} )
+		).toBeVisible();
+
+		await expect(
+			blockSettings.getByRole( 'tab', {
+				name: 'Settings',
+				selected: true,
+			} )
+		).toBeVisible();
+
+		await expect(
+			blockSettings
+				.getByRole( 'tabpanel', {
+					name: 'Settings',
+				} )
+				.getByRole( 'heading', {
+					name: 'Link Settings',
+				} )
+		).toBeVisible();
+
+		// Click the back button to go back to the Nav block.
+		await blockSettings
+			.getByRole( 'button', {
+				name: 'Go to parent Navigation block',
+			} )
+			.click();
+
+		// Check we're back on the Nav block list view.
+		const listViewPanel = page.getByRole( 'tabpanel', {
+			name: 'List View',
+		} );
+
+		await expect( listViewPanel ).toBeVisible();
+	} );
+
+	test( `can add submenus`, async ( {
+		admin,
+		page,
+		editor,
+		requestUtils,
+	} ) => {
+		await admin.createNewPost();
+		await requestUtils.createNavigationMenu( {
+			title: 'Test Menu',
+			content: `<!-- wp:navigation-link {"label":"Top Level Item 1","type":"page","id":250,"url":"http://localhost:8888/quod-error-esse-nemo-corporis-rerum-repellendus/","kind":"post-type"} /-->
+			<!-- wp:navigation-submenu {"label":"Top Level Item 2","type":"page","id":250,"url":"http://localhost:8888/quod-error-esse-nemo-corporis-rerum-repellendus/","kind":"post-type"} -->
+				<!-- wp:navigation-link {"label":"Test Submenu Item","type":"page","id":270,"url":"http://localhost:8888/et-aspernatur-recusandae-non-sint/","kind":"post-type"} /-->
+			<!-- /wp:navigation-submenu -->`,
+		} );
+
+		await editor.insertBlock( { name: 'core/navigation' } );
+
+		await editor.openDocumentSettingsSidebar();
+
+		const listViewTab = page.getByRole( 'tab', {
+			name: 'List View',
+		} );
+
+		await listViewTab.click();
+
+		const listView = page.getByRole( 'treegrid', {
+			name: 'Block navigation structure',
+			description: 'Structure for navigation menu: Test Menu',
+		} );
+
+		// click on options menu for the first menu item and select remove.
+		const firstMenuItem = listView
+			.getByRole( 'gridcell', {
+				name: 'Page Link link',
+			} )
+			.filter( {
+				hasText: 'Block 1 of 2, Level 1', // proxy for filtering by description.
+			} );
+
+		// Focus the node to reveal the "3 dots" options menu button.
+		const firstMenuItemAnchor = listView.getByRole( 'link', {
+			name: 'Top Level Item 1',
+			includeHidden: true,
+		} );
+		await firstMenuItemAnchor.focus();
+
+		// The options menu button is a sibling of the menu item gridcell.
+		const firstItemOptions = firstMenuItem
+			.locator( '..' ) // parent selector.
```

Comment:
TIL! neat

Reply:
Same. Turns out reading the docs is useful 😆

Reply:
For whatever reason that didn't work. It ended up producing

```
getByRole( 'treegrid', { name: 'Block navigation structure' } )
	.getByRole( 'row' )
	.filter( {
		has: getByRole( 'treegrid', { name: 'Block navigation structure' } )
			.getByRole( 'gridcell', { name: 'Page Link link' } )
			.filter( { hasText: 'Block 1 of 2, Level 1' } ),
	} )
	.getByRole( 'button', { name: 'Options for Page Link block' } );
```